source/core/module_typeHelpers.js

```diff
@@ -0,0 +1,340 @@
+            var typeRef = valueRef.typeRef;
+            var sizeOfFloat = Module.typeHelpers.topAQSize(typeRef);
+            var visitFn;
+            var fnName = '';
+            switch (sizeOfFloat) {
+            case 4:
+                visitFn = typeVisitor.visitSingle;
+                fnName = 'visitSingle';
+                break;
+            case 8:
+                visitFn = typeVisitor.visitDouble;
+                fnName = 'visitDouble';
+                break;
+            default:
+                throw new Error('Unexpected size for a Float value. Found: ' + sizeOfFloat);
+            }
+
+            validateVisitMethod(visitFn, fnName);
+            return visitFn.call(typeVisitor, valueRef, data);
+        };
+
+        var dispatchVisitString = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitString, 'visitString');
+            return typeVisitor.visitString(valueRef, data);
+        };
+
+        var dispatchVisitComplex = function (typeVisitor, valueRef, data) {
+            var typeRef = valueRef.typeRef,
+                sizeOfComplex = Module.typeHelpers.topAQSize(typeRef),
+                visitFn,
+                fnName;
+            switch (sizeOfComplex) {
+            case 8:
+                visitFn = typeVisitor.visitComplexSingle;
+                fnName = 'visitComplexSingle';
+                break;
+            case 16:
+                visitFn = typeVisitor.visitComplexDouble;
+                fnName = 'visitComplexDouble';
+                break;
+            default:
+                throw new Error('Unexpected size for a Complex value. Found: ' + sizeOfComplex);
+            }
+
+            validateVisitMethod(visitFn, fnName);
+            return visitFn.call(typeVisitor, valueRef, data);
+        };
+
+        var dispatchVisitAnalogWaveform = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitAnalogWaveform, 'visitAnalogWaveform');
+            return typeVisitor.visitAnalogWaveform(valueRef, data);
+        };
+
+        var dispatchVisitTimestamp = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitTimestamp, 'visitTimestamp');
+            return typeVisitor.visitTimestamp(valueRef, data);
+        };
+
+        var dispatchVisitPath = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitPath, 'visitPath');
+            return typeVisitor.visitPath(valueRef, data);
+        };
+
+        var dispatchVisitArray = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitArray, 'visitArray');
+            return typeVisitor.visitArray(valueRef, data);
+        };
+
+        var dispatchVisitCluster = function (typeVisitor, valueRef, data) {
+            validateVisitMethod(typeVisitor.visitCluster, 'visitCluster');
+            return typeVisitor.visitCluster(typeVisitor, valueRef, data);
```

Comment:
Should be `typeVisitor.visitCluster(valueRef, data);`

Reply:
Yes, I'lll fix it